### PR TITLE
NXBT-2973: Introduce a builder-base

### PR DESCRIPTION
**Pull request by kevinleturc**

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -57,13 +57,19 @@ pipeline {
...
+              skaffold build -f builder-java/skaffold.yaml~gen
+              
               envsubst < builder-java-firefox/skaffold.yaml > builder-java-firefox/skaffold.yaml~gen
               skaffold build -f builder-java-firefox/skaffold.yaml~gen
 
               envsubst < builder-java-firefox-nodejs/skaffold.yaml > builder-java-firefox-nodejs/skaffold.yaml~gen
               skaffold build -f builder-java-firefox-nodejs/skaffold.yaml~gen
+              
+              envsubst < builder-rpm/skaffold.yaml > builder-rpm/skaffold.yaml~gen
+              skaffold build -f builder-rpm/skaffold.yaml~gen
```
Comment (ataillefer) [inline Jenkinsfile]: Shouldn't we do the same for master?
Also, do yo think we could factorize `envsubst` and `skaffold` with functions?

File: builder-base/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+LABEL description=${DESCRIPTION}
+LABEL scm-url=${SCM_REPOSITORY}
+LABEL version=${VERSION}
+LABEL scm-ref=${SCM_REF}
+
+ARG SKAFFOLD_VERSION
+
+# Cannot run pipelines with a non root user yet, see https://jira.nuxeo.com/browse/NXBT-2892
+#ARG JENKINS_USER=jenkins
```
Comment (ataillefer) [inline builder-base/Dockerfile]: Then the jenkins user (commented) lines could be removed from all other Dockerfiles right?

File: builder-java/skaffold.yaml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+apiVersion: skaffold/v1beta11
+kind: Config
+build:
+  tagPolicy:
+    envTemplate:
+      template: "{{.DOCKER_REGISTRY}}/{{.ORG}}/{{.IMAGE_NAME}}:{{.VERSION}}"
+  artifacts:
+    - image: builder-java8
```
Comment (ataillefer) [inline builder-java/skaffold.yaml]: Shouldn't this be done in a separate commit with NXBT-2974?

File: builder-base/Dockerfile (hunk 2 above)
Comment (kevinleturc) [reply builder-base/Dockerfile]: I wasn't certain about that because in children image we'll need to be root to install some stuff, no?

File: builder-java/skaffold.yaml (hunk 3 above)
Comment (kevinleturc) [reply builder-java/skaffold.yaml]: It is :).

File: builder-base/Dockerfile (hunk 2 above)
Comment (ataillefer) [reply builder-base/Dockerfile]: Yep, we'll see when/if we can use a jenkins user

**Comment by kevinleturc** [comment]
I pushed changes after @ataillefer review.
@bdelbosc / @ataillefer can you review again please?

**Comment by kevinleturc** [comment]
The builder is currently under tests in https://github.com/nuxeo/nuxeo/pull/3297
